src/components/SerialConsole/SerialConsole.js

```diff
@@ -0,0 +1,185 @@
+            onData={this.props.onData}
+            onTitleChanged={this.props.onTitleChanged}
+            onResize={this.props.onResize}
+          />
+        );
+        disconnectDisabled = '';
+        break;
+      case DISCONNECTED:
+        terminal = (
+          <EmptyState>
+            <EmptyState.Title>
+              {this.props.textDisconnectedTitle}
+            </EmptyState.Title>
+            <EmptyState.Info>{this.props.textDisconnected}</EmptyState.Info>
+            <EmptyState.Action>
+              <Button
+                bsStyle="primary"
+                bsSize="large"
+                onClick={this.props.onConnect}
+              >
+                {this.props.textConnect}
+              </Button>
+            </EmptyState.Action>
+          </EmptyState>
+        );
+        break;
+      case LOADING:
+      default:
+        terminal = <span>{this.props.textLoading}</span>;
+        break;
+    }
+
+    return (
+      <div className={topClassName} id={id}>
+        <div className="console-actions">
+          <button
+            ref="disconnectButton"
+            id={`${idPrefix}-disconnect`}
+            className={`btn btn-default console-actions-buttons ${disconnectDisabled}`}
+            onClick={this.onDisconnectClick}
+          >
+            {this.props.textDisconnect}
+          </button>
+
+          <button
+            ref="resetButton"
+            id={`${idPrefix}-reconnect`}
+            className="btn btn-default console-actions-buttons"
+            onClick={this.onResetClick}
+          >
+            {this.props.textReconnect}
+          </button>
+        </div>
+
+        <div className="panel-body console-terminal">{terminal}</div>
+      </div>
+    );
+  }
+}
+
+SerialConsole.propTypes = {
+  /** Initiate connection to backend. In other words, the calling components manages connection state. */
+  onConnect: PropTypes.func.isRequired,
+  /** Close connection to backend */
+  onDisconnect: PropTypes.func.isRequired,
+  /** Terminal has been resized, backend shall be informed. (rows, cols) => {} */
+  onResize: PropTypes.func,
+  /** Terminal produced data, like key-press */
+  onData: PropTypes.func,
+  /** Terminal title has been changed. */
+  onTitleChanged: PropTypes.func,
+
+  /** Connection status, a value from [''connected', 'disconnected', 'loading']. Default is 'loading' for a not matching value. */
+  status: PropTypes.string.isRequired,
+  id: PropTypes.string,
+
+  /** Enable customization */
+  topClassName: PropTypes.string,
+
+  textDisconnect: PropTypes.string,
+  textDisconnectedTitle: PropTypes.string,
+  textDisconnected: PropTypes.string,
+  textLoading: PropTypes.string,
+  textReconnect: PropTypes.string,
+  textConnect: PropTypes.string
+};
+
+SerialConsole.defaultProps = {
+  topClassName: '',
+
+  id: '',
+
+  onTitleChanged: title => {},
+  onData: data => {},
+  onResize: (rows, cols) => {},
+  onDisconnect: () => {},
+  onConnect: () => {},
```

Comment:
These could all be set to `helpers.noop`

Reply:
done